packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -30,43 +32,64 @@ export default class AssetGraphBuilder extends EventEmitter {
   requestGraph: RequestGraph;
   queue: PromiseQueue;
   controller: AbortController;
-  changedAssets: Map<string, Asset>;
+  changedAssets: Map<string, Asset> = new Map();
+  options: ParcelOptions;
+  cacheKey: string;
+  cache: Cache;
+
+  async init({config, options, entries, targets, assetRequest}: Opts) {
+    this.options = options;
+    let {minify, hot, scopeHoist} = options;
+    this.cacheKey = md5FromObject({
+      options: {minify, hot, scopeHoist},
+      entries,
+      targets
```

Comment:
Would it make sense to eventually have different graphs per target? That way we wouldn't lose the entire graph when we add or remove a target? Seems like we still have to keep entries together in one graph since we might end up with shared bundles between entries, but targets seem like they could maybe be separated cleanly?

Also, I've been thinking that options could maybe be treated similarly to files. Requests could specify which options they care about like they specify which files they care about right now, and when they change we can invalidate only the nodes that are connected to those options. That way we don't have to resolve everything all over when the minify option is changed.

Reply:
Yeah that's an interesting idea. Currently, we would still get some caching benefit since the individual assets are still cached in addition to the graph, so only the new target would actually be compiled. Not sure how often targets will be added/removed, but it's something to consider.